### PR TITLE
Update/fix development installation docs

### LINKED ISSUE #532: Docs: Streamline BEAST Development page
The BEAST Development page contains duplicate "Development Install" sections as well as an empty "Collaborating and Contributing" section. Moreover, it might be nice to add a "Submit Pull Request" section/blurb, and also a "Keeping up-to-date with upstream master branch" (see #529).

### PR DESCRIPTION
**Pull request by jwuphysics**
Fixes #532 and #529.

This PR updates the text for the BEAST development install guide. At some point, the installation instructions were removed and then partially restored; now they are more coherent and streamlined. I've also included short summaries on how to submit a pull request (collaborating/contributing) and how to fetch the upstream master branch (staying up to date). 

`build_docs` passed locally for this document.

### CODE REVIEW

File: docs/beast_development.rst [hunk 1]
```diff
@@ -42,35 +42,51 @@ project is:
...
 
 Adding Branches
 ===============
 
 - Make sure you are in the directory for your fork of the beast. You will be on
-  branch 'master' by default.
+  branch `master` by default.
 
-- Create and switch to a branch (here named 'beast-dev1'; generally it's good
+- Create and switch to a branch (here named `beast-dev1`; generally it's good
```
Comment (karllark) [inline docs/beast_development.rst:89]: This sentence does not seem to relate to the code-block below.  Is there something missing?

File: docs/beast_development.rst [hunk 2]
```diff
@@ -211,8 +207,53 @@ The resulting HTML files are placed in `docs/_build/html` subdirectory, and
...
+     $ git checkout master
+
+- Fetch the project's up-to-date distribution:
+
+  .. code-block:: console
+
+     $ git fetch upstream
+
+- If there are any differences, then merge the official `master` branch (upstream)
+  with your fork's `master` (you may need to write a commit message):
```
Comment (karllark) [inline docs/beast_development.rst:245]: How does one check if there are differences?  I usually just run the `git merge` command to find out.

File: docs/beast_development.rst [hunk 3]
```diff
@@ -211,8 +207,53 @@ The resulting HTML files are placed in `docs/_build/html` subdirectory, and
...
 
-Collaborating and Contributing
-==============================
+Submitting a Pull Request
+=========================
+
+Once you have changes that you'd like to contribute back to the project or share
+with collaborators, you can open a pull request. It is a good idea to check with
+the projects or your collaborators which branch of their BEAST repository you
+should send the pull requests.
```
Comment (karllark) [inline docs/beast_development.rst:216]: Isn't it always the BEAST-Fitting/beast master branch?

File: docs/beast_development.rst (hunk 2 above)
Comment (jwuphysics) [reply docs/beast_development.rst:245]: That's a good point, although it's possible to fetch upstream and receive no new updates.

File: docs/beast_development.rst (hunk 1 above)
Comment (jwuphysics) [reply docs/beast_development.rst:89]: I think the view here jumps from lines 92 to 113, it actually is followed by the code block
```
  .. code-block:: console

     $ git checkout -b beast-dev1
```

File: docs/beast_development.rst (hunk 3 above)
Comment (jwuphysics) [reply docs/beast_development.rst:216]: True, I just just re-added this text. Will clarify.

**Comment by jwuphysics** [comment]
Implemented some small changes as requested. @mdecleir do you want to take a look at the guide/docs?

File: docs/beast_development.rst [hunk 4]
```diff
@@ -42,97 +42,85 @@ project is:
 Development Install
 ===================
 
-It is much easier to perform development if any changes you make to the code are
-immediately reflected in how the ``beast`` runs (in contrast to needed to perform a
-new install each team). This can be achieved by using a development install.
+It is much easier to develop the code if any changes you make are immediately
+reflected in how the ``beast`` runs (in contrast to performing a new install each
+team). This can be achieved by using a development install. First, create a fork
```
Comment (mdecleir) [inline docs/beast_development.rst:47]: team -->time

File: docs/beast_development.rst [hunk 5]
```diff
@@ -42,97 +42,85 @@ project is:
...
+``beast`` repository that contains `setup.py`, and run:
 
-  .. code-block:: console
+.. code-block:: console
 
-     $ git checkout -b beast-dev1
+   $ pip install -e .
 
-- Instead, if you want to create first a branch and then switch to it:
+Alternatively, you can perform a development install directly though Python
```
Comment (mdecleir) [inline docs/beast_development.rst:75]: though--> through

**Comment by mdecleir** [comment]
This looks really great!! Especially the part about staying up-to-date is really useful, cause this always confused me in the beginning! The only thing you might want to add is the "git diff" which I use all the time to see what I actually changed.

**Comment by mdecleir** [comment]
Not sure if you also want to add a section about deleting a branch if you are done (after it has been merged etc), to keep things clean?

File: docs/beast_development.rst (hunk 4 above)
Comment (jwuphysics) [reply docs/beast_development.rst]: I just ended up simplifying this part -- thanks though!.

**Comment by jwuphysics** [comment]
> Not sure if you also want to add a section about deleting a branch if you are done (after it has been merged etc), to keep things clean?

Good point, although Github provides the option to automatically do that at the bottom of a PR. In my opinion it's not bad to have old branches lying around, unless you have a ton of them -- and people who are regular contributors to the BEAST probably know how to get rid of them :smile:

**Comment by jwuphysics** [comment]
@karllark Changes implemented as suggested. Also thanks to @mdecleir for reviewing!

**Pull request merged**